torch/csrc/jit/export.cpp

```diff
@@ -66,7 +72,24 @@ void encodeTensor(onnx::TensorProto * p, const at::Tensor & tensor) {
   }
   p->set_data_type(onnx_type);
   // CPU's HalfTensor doesn't have contiguous(), so first calling contiguous()
-  p->set_raw_data(tensor.contiguous().toBackend(at::kCPU));
+  auto t = tensor.contiguous().toBackend(at::kCPU);
+  // Add a buffer to the raw_data_export_map for the caller to dump into an
+  // external data store. If external_ref is not specified, we instead dump
+  // the contiguous data into the protobuf itself
+  if (external_ref) {
+    // For now, we use the name of the tensor as the external lookup name to
+    // avoid ONNX protobuf changes.
+    JIT_ASSERT(external_ref.value() == p->get_name());
+    JIT_ASSERT(raw_data_export_map != nullptr);
+    JIT_ASSERT(raw_data_export_map->count(external_ref.value()) == 0);
+    size_t copy_bytes = t.type().elementSizeInBytes() * t.numel();
+    (*raw_data_export_map)[external_ref.value()] =
+        std::string(copy_bytes, '\0');
```

Comment:
Do you really want to zero initialize the string?

Reply:
Removing